schedule/schedule-appengine/src/test/java/com/jasify/schedule/appengine/spi/ActivityEndpointTest.java

```diff
@@ -0,0 +1,517 @@
+            jasAddActivityTypeRequest.setActivityType(new ActivityType());
+            jasAddActivityTypeRequest.setOrganizationId(organizationKey);
+
+            endpoint.addActivityType(newCaller(55, true), jasAddActivityTypeRequest);
+        } finally {
+            testOrganizationServiceFactory.tearDown(false);
+        }
+    }
+
+    @Test(expected = BadRequestException.class)
+    public void testAddActivityTypeThrowsBadRequestException() throws Exception {
+        try {
+            testOrganizationServiceFactory.setUp();
+            OrganizationService organizationService = OrganizationServiceFactory.getOrganizationService();
+            Key organizationKey = Datastore.allocateId(Organization.class);
+            Organization organization = new Organization();
+            expect(organizationService.getOrganization(organizationKey)).andReturn(organization);
+            testOrganizationServiceFactory.replay();
+
+            ActivityService activityService = ActivityServiceFactory.getActivityService();
+            JasAddActivityTypeRequest jasAddActivityTypeRequest = new JasAddActivityTypeRequest();
+            jasAddActivityTypeRequest.setActivityType(new ActivityType());
+            jasAddActivityTypeRequest.setOrganizationId(organizationKey);
+            activityService.addActivityType(organization, jasAddActivityTypeRequest.getActivityType());
+            expectLastCall().andThrow(new FieldValueException(""));
+            testActivityServiceFactory.replay();
+
+            endpoint.addActivityType(newCaller(55, true), jasAddActivityTypeRequest);
+        } finally {
+            testOrganizationServiceFactory.tearDown(false);
+        }
+    }
+
+    @Test(expected = NotFoundException.class)
+    public void testAddActivityTypeThrowsNotFoundExceptionOnAddId() throws Exception {
+        try {
+            testOrganizationServiceFactory.setUp();
+            OrganizationService organizationService = OrganizationServiceFactory.getOrganizationService();
+            Key organizationKey = Datastore.allocateId(Organization.class);
+            Organization organization = new Organization();
+            expect(organizationService.getOrganization(organizationKey)).andReturn(organization);
+            testOrganizationServiceFactory.replay();
+
+            ActivityService activityService = ActivityServiceFactory.getActivityService();
+            JasAddActivityTypeRequest jasAddActivityTypeRequest = new JasAddActivityTypeRequest();
+            ActivityType activityType = new ActivityType();
+            jasAddActivityTypeRequest.setActivityType(activityType);
+            jasAddActivityTypeRequest.setOrganizationId(organizationKey);
+            expect(activityService.addActivityType(organization, activityType)).andReturn(organizationKey);
+            expect(activityService.getActivityType(organizationKey)).andThrow(new EntityNotFoundException());
```

Comment:
I don't understand why you had to say `tearDown(false);`  
How will you know if the methods get called?